pkg/virt-handler/vm.go

```diff
@@ -557,6 +557,12 @@ func (d *VirtualMachineController) updateVMIStatus(vmi *v1.VirtualMachineInstanc
 
 			existingInterfacesSpecByName := map[string]v1.Interface{}
 			for _, existingInterfaceSpec := range vmi.Spec.Domain.Devices.Interfaces {
+				if existingInterfaceSpec.MacAddress != "" {
```

Comment:
The whole function is one big mess, doing endless operation and requiring a reader to heavily invest in all possible details. Looking at the unit tests, it is even worse, I am not able to detect what is really tested/covered.

Personally, I would not have touched this function before extracting out of it the network part and refactor it in a way that it uses proper functions that can explain what is done.

If I look at this specific addition, then the logic is hijacking a block of code which was responsible to index the spec interfaces by their name.
Adding it in here, makes it even harder to understand what is going on.
Please use a separate loop block, or better, a function that loops the needed and mutates the domain interface status. If you can avoid mutating inside the function, even better.

Something like this:
```
for mac, name := range indexSpecIfaceNamesByMac(vmi) { // Returns a map for non empty iface names.
    if domainIfaceSatus, exists := domainInterfaceStatusByMac[mac]; exists {
        domainIfaceSatus.Name = name
        domainInterfaceStatusByMac[mac] = domainIfaceSatus
    }
}
```

The above is by itself a function that does one thing: `setMissingDoaminInterfacesStatusNames`

Reply:
is it ok then that i will add it in the loop that exists right about it and creates the map of the mac?

see please for this discussion
https://github.com/kubevirt/kubevirt/pull/4335#discussion_r501749430

Reply:
I think there is no loop that you can hack into to do this. All other blocks I see here are exercising an indexing operation, adding anything to them will just make it hard to understand what is going on.

Reply:
https://github.com/kubevirt/kubevirt/pull/4335/files#diff-7a3f9e0461f554b07ac195375eb0f8db7d46579071795a65c459c721b1a8204aR555

this one i can, add for in for as in 
https://github.com/kubevirt/kubevirt/pull/4335#discussion_r501749430

but i will do in two loops if you prefer, on it

Reply:
Done